### ISSUE TITLE
For some runs, some parameters (ex: seasons) from the Python input file cannot overwrite values from the default cfgs

### ISSUE THREAD

**Issue by zshaheen** [E3SM-Project/acme_diags#132]
## Problem
In the default cfgs/jsons, we sometimes have the runs for a specific season(s). See the cfg snippets below.
For the seasons parameter, and maybe more in the future, we can't have the values in the *py overwrite these parameters. It causes the wrong plots to be created.
 
## Why it breaks
In [this comment](https://github.com/ACME-Climate/acme_processflow/issues/41#issue-260719818), you gave one example of plots that break in the acme_processflow, but not when running acme_diags independently.
Below are the parameters for the SWCF variable:
```cfg
[#]
sets = ["lat_lon"]
case_id = "CERES-EBAF-TOA-v2.8"
variables = ["SWCF"]
ref_name = "ceres_ebaf_toa_v2.8"
reference_name = "CERES-EBAF Jan 2000-Dec 2015"
seasons = ["ANN"]
regions = ["global"]
contour_levels = [-120, -110, -100, -90, -80, -70, -60, -50, -40, -30, -20, -10, 0]
diff_levels = [-60, -50, -40, -30, -20, -10, -5, 5, 10, 20, 30, 40, 50, 60]
 
[#]
sets = ["lat_lon"]
case_id = "CERES-EBAF-TOA-v2.8"
variables = ["SWCF"]
ref_name = "ceres_ebaf_toa_v2.8"
reference_name = "CERES-EBAF Jan 2000-Dec 2015"
seasons = ["DJF", "MAM", "JJA", "SON"]
regions = ["global"]
contour_levels = [-180, -160, -140, -120, -100, -80, -60, -40, -20,  0]
diff_levels = [-60, -50, -40, -30, -20, -10, -5, 5, 10, 20, 30, 40, 50, 60]
```

**When you run acme_diags independently, you probably didn’t put `seasons=['DJF', 'MAM', 'JJA', 'SON', 'ANN']` in the `*py` file (as in `acme_diags_driver.py -p something.py`)**
**So the seasons parameter wasn’t overwritten, so the plots with the correct contour levels were created.**

---

Now when you do put `seasons = ['DJF', 'MAM', 'JJA', 'SON', 'ANN']` in the `*py` file, **as the processflow does**, you’ll get the following parameters for SWCF and others:
```cfg
[#]
sets = ["lat_lon"]
case_id = "CERES-EBAF-TOA-v2.8"
variables = ["SWCF"]
ref_name = "ceres_ebaf_toa_v2.8"
reference_name = "CERES-EBAF Jan 2000-Dec 2015"
seasons = ["DJF", "MAM", "JJA", "SON", "ANN"]  # This is bad
regions = ["global"]
contour_levels = [-120, -110, -100, -90, -80, -70, -60, -50, -40, -30, -20, -10, 0]
diff_levels = [-60, -50, -40, -30, -20, -10, -5, 5, 10, 20, 30, 40, 50, 60]
 
[#]
sets = ["lat_lon"]
case_id = "CERES-EBAF-TOA-v2.8"
variables = ["SWCF"]
ref_name = "ceres_ebaf_toa_v2.8"
reference_name = "CERES-EBAF Jan 2000-Dec 2015"
seasons = ["DJF", "MAM", "JJA", "SON", "ANN"]  # This is also bad
regions = ["global"]
contour_levels = [-180, -160, -140, -120, -100, -80, -60, -40, -20,  0]
diff_levels = [-60, -50, -40, -30, -20, -10, -5, 5, 10, 20, 30, 40, 50, 60]
```

**You can see now that this is why you get the wrong ANN plot with the wrong contour levels, it’s created from the second set of parameters.
Even when you use multiprocessing, the order is even more random. Whatever parameter is done last, that’s the one that’s saved by overwriting the previous one.**

## Possible Solutions
* have some sort of way in the jsons/cfgs of telling what parameters to not overwrite. So a meta-parameter like `params_to_not_overwrite=['seasons']` in the two aforementioned parameter sets.
* @chengzhuzhang's idea, have some way of telling cdp that for some parameters, don't prioritize the ones from the *py. In this case for the two aforementioned cfg files, we have to prioritize the ones from the cfgs/jsons.

**Comment by zshaheen** [comment]
The actual issue was discovered on [ACME-Climate/acme_processflow/41](https://github.com/ACME-Climate/acme_processflow/issues/41).

**Comment by zshaheen** [comment]
@golaz's suggestion is to have parameters such as `seasons` be 'selectors' or something like that. So when `seasons = ['ANN']`, it won't replace the `seasons` in all the default runs with `['ANN']`. Instead, it would select all those diags that had `'ANN'` for the `seasons`. So given the following three runs:
```cfg
[run 1]
...
seasons = ['ANN']
...

[run 2]
...
seasons = ['ANN', 'DJF', 'MAM', 'JJA', 'SON']
...

[run 3]
...
seasons = ['DJF', 'MAM', 'JJA', 'SON']
...
```

the selected runs would be:

```cfg
[run 1]
...
seasons = ['ANN']
...

[run 2]
...
seasons = ['ANN']
...
```

Note that `run 3` wasn't selected.

**Comment by zshaheen** [comment]
Done in the `selector` branch, will be merged by #181.
